restli-common/src/main/java/com/linkedin/restli/common/validation/RestLiDataValidator.java

```diff
@@ -106,7 +106,7 @@
 
   // ReadOnly fields are treated as optional for these types of requests
   private static final Set<ResourceMethod> readOnlyOptional = new HashSet<>(Arrays.asList(
```

Comment:
Is this mapping even needed anymore? If you look at the logic path, `#validateInput(RecordTemplate)` already has a switch to check that the resource method is one of `[create, batch_create, update, batch_update]`, so it seems like the "read-only optional" logic is applied for all "input" method types. partial_update and delete have their own separate code paths.

Reply:
Ack, fixed it.